src/rdsn/src/utils/metrics.h

```diff
@@ -389,53 +389,55 @@ template <typename T, typename = typename std::enable_if<std::is_arithmetic<T>::
 class gauge : public metric
```

Comment:
Why didn't you change `typename T` to `typename value_type` in the above line?

Reply:
> Why didn't you change `typename T` to `typename value_type` in the above line?

`value_type` may be a combination of multiple template parameters. For example, in std::map, `value_type` is defined as `std::pair<const _Key, _Tp>`. Thus customarily `value_type` is used to decouple from template parameters though for `gauge` it is just the same.